am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -118,6 +119,64 @@ void whenRevokingResourceAccessThatDoesNotExistNoErrorExpected() {
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceWithNonExistentRelationshipShouldNotRemoveAnyRecordFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "");
+
+        service.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(accessorId)
+            .accessorType(USER)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute(ROOT_ATTRIBUTE)
+            .securityClassification(PUBLIC)
+            .relationship("NonExistentRelationship")
+            .build());
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(1);
```

Comment:
Please change this to have specific match. When we start randomise data counting won't be enough.

Reply:
👍